src/map/ai/states/item_state.cpp

```diff
@@ -85,7 +85,7 @@ CItemState::CItemState(CCharEntity* PEntity, uint16 targid, uint8 loc, uint8 slo
         throw CStateInitException(std::move(m_errorMsg));
```

Comment:
teschnei 3 hours ago Owner
@teschnei 
>we did something similar in magic_state, but it would cause a crash because sometimes PTarget was null - can you do PTarget ? PTarget : m_PEntity instead of just PTarget?

see line above, github wont let me mark it for comment

Reply:
wew guess that was shit also, what i meant to point to was 
https://github.com/metalfiiish/darkstar/blob/8dd93a4a4686265f36202ebae5c1e1befb03fd8e/src/map/ai/states/item_state.cpp#L83

Reply:
So shouldn't I just do what Lotus said and make it do an or 'PTarget ? PTarget : m_PEntity instead of just PTarget'?

Reply:
no cause PTarget would never be null where he told you to change it since the state throws an exception if it is in the line i linked

Reply:
So I am a bit lost now on who's changes to do to get this PR slated. A cliff notes for what you two want for me to do exactly?

Reply:
you can ignore the stuff about PTarget, but would be best to move
`Lunar<CLuaBaseEntity>::push(LuaHandle, &LuaBaseEntityCaster);`
to after `lua_pushinteger(L, param);` so we dont need to run any mass replaces